File: elixir/nth-prime/nth_prime.exs
defmodule Prime do

  @doc """
  Generates the nth prime.
  """
  @spec nth(non_neg_integer) :: non_neg_integer
  def nth(count) do
    if count <= 0, do: raise ArgumentError
    2 |> Stream.unfold(&{&1, &1+1})
      |> Stream.scan([], fn(candidate, primes) ->
           if Enum.any?(primes, &rem(candidate, &1)==0) do
             primes
           else
             [candidate | primes]
           end
         end)
      |> Enum.find(&length(&1)==count)
      |> Enum.at(0)
  end
end

File: ruby/nth-prime/nth_prime.rb
class Prime
  def self.nth(n)
    first(n).last
  end

  def self.first(n)
    raise ArgumentError if n <= 0
    primes = [2]
    k = 3
    while primes.size < n
      primes << k unless primes.any?{|p| k % p == 0 }
      k += 2
    end
    primes
  end
end
